bgzf.c

```diff
@@ -273,9 +273,7 @@ static int bgzf_idx_flush(BGZF *fp) {
     hts_idx_cache_entry *e = mt->idx_cache.e;
     int i;
 
-    assert(mt->idx_cache.nentries == 0 || mt->block_written >= e[0].block_number);
-
-    for (i = 0; i < mt->idx_cache.nentries && e[i].block_number == mt->block_written; i++) {
+    for (i = 0; i < mt->idx_cache.nentries && e[i].block_number <= mt->block_written; i++) {
```

Comment:
The value `(mt->block_address << 16) + e[i].offset` is only valid when `e[i].block_number == mt->block_written` otherwise an incorrect index entry would be written.  Which suggests that using `==` in the for loop was correct, and the assertion should have been to check that blocks haven't been written without adding their index entries, i.e.:
```
assert(mt->idx_cache.nentries == 0 || mt->block_written <= e[0].block_number)
```

I don't think this can happen, but then isn't that the sort of reasoning that the assertion was trying to check?